openweathermap-ios/openweathermap-ios/Maps/ViewControllers/MapViewController.swift

```diff
@@ -57,6 +63,12 @@ extension MapViewController: MKMapViewDelegate {
     }
     
     func mapView(_ mapView: MKMapView, regionDidChangeAnimated animated: Bool) {
+        let newCoordinate = mapView.centerCoordinate
+        let distance = newCoordinate.betweenTwoPoints(coordinate: lastCenterCoordinate)
+        if distance > 20_000 {
+            lastCenterCoordinate = newCoordinate
+            weatherManager.fetchWeather(latitude: newCoordinate.latitude, longitude: newCoordinate.longitude)
+        }
         updateAnnotation(location: mapView.centerCoordinate)
```

Comment:
This after obtaining the result from the server.

Reply:
I didn't get it.

Reply:
You're right. This note has no information.
I'll fix it :).

Reply:
My suggestion to add the annotation is to make changes to the WeatherManagerDelegate.

But to update add the annotation it was necessary to use Dispatch.

This is the best solution?
```
extension MapViewController: WeatherManagerDelegate {
    func didUpdateWeather(weather: WeatherModel) {
        DispatchQueue.main.async {
            self.cityName = "\(weather.city)"
            self.descriptionWeather = "\(weather.description) (\(weather.temperature))"
            self.updateAnnotation(location: self.lastCenterCoordinate)
        }
    }
}
```

Reply:
"Best Solution" is too strong. Be happy with a "Solution".

Reply:
It does what you need. But know you fix the logic, it's clear you don't need to keep states on ViewController. So, consider removing the properties `cityName` and `descriptionWeather` from the Class scope, and consider passing those values by function parameter.

Obs.: I know I commented before to remove the parameters from the function. I was because your code was different and in that case the parameters was useless. But know, the properties are useless.

Reply:
The same with the `lastCenterCoordinate`. Why are you passing something that you have available on the class scope? 

Consider removing the parameter `location` and access the lastCenterCoordinate directly from the self.

Reply:
I will make the changes. Suggestions are always welcome and I usually do as suggested.

Reply:
No problem, I'm here to help.

Reply:
I removed the parameters from the function.
`updateAnnotation()`

Because when I move on the map I remove the annotation and if I don't update data on the server I add the annotation on the map again.
Obs.: when you move the map it shows the target and when to add a annotation.

```
extension MapViewController: WeatherManagerDelegate {
    func didUpdateWeather(weather: WeatherModel) {
        DispatchQueue.main.async {
            self.cityName = "\(weather.city)"
            self.descriptionWeather = "\(weather.description) (\(weather.temperature))"
            self.updateAnnotation()
        }
    }
}
```
And a control to keep the note if it is not updated.

```
if distance < 20_000 {
            updateAnnotation()
        }
```
I believe it just got better.